aiida_cp2k/calculations/__init__.py

```diff
@@ -415,3 +451,29 @@ def _atoms_to_xyz(atoms):
     xyz = f"{len(elem_coords)}\n\n"
     xyz += "\n".join(map(add, elem_symbols, elem_coords))
     return xyz
+
+
+def _trajectory_to_xyz_and_cell(trajectory, atoms):
+    """Converts postions and cell from a TrajectoryData  to string, taking care of element tags from ASE atoms.
+
+    :param atoms: ASE Atoms instance
+    :param trajectory: TrajectoryData instance
+    :returns: positions str (in xyz format) and cell str
+    """
+    cell = "#   Step   Time [fs]       Ax [Angstrom]       Ay [Angstrom]       Az [Angstrom]       Bx [Angstrom]       By [Angstrom]       Bz [Angstrom]       Cx [Angstrom]       Cy [Angstrom]       Cz [Angstrom]      Volume [Angstrom^3]\n"
+    xyz = ""
+    elem_symbols = kind_names(atoms)
+
+    for i, step in enumerate(trajectory.get_array("positions")):
+        elem_coords = [f"{p[0]:25.16f} {p[1]:25.16f} {p[2]:25.16f}" for p in step]
+        xyz += f"{len(elem_coords)}\n"
+        xyz += f"i = {i+1} , time = {(i+1)*0.5} \n"
+        xyz += "\n".join(map(add, elem_symbols, elem_coords))
+        xyz += "\n"
+    if "cells" in trajectory.get_arraynames():
+        cell_vecs = [
+            f"{i+1} {(i+1)*0.5:6.3f} {p[0][0]:25.16f} {p[0][1]:25.16f} {p[0][2]:25.16f} {p[1][0]:25.16f} {p[1][1]:25.16f} {p[1][2]:25.16f} {p[2][0]:25.16f} {p[2][1]:25.16f} {p[2][2]:25.16f} {np.dot(p[0],np.cross(p[1],p[2]))}"
+            for (i, p) in enumerate(trajectory.get_array("cells"))
+        ]
+        cell += "\n".join(cell_vecs)
```

Comment:
Rewrite this function. The `TrajectoryData` class has a method to get the structure from from a given frame.

Reply:
I guess this is done in [06a33f3](https://github.com/aiidateam/aiida-cp2k/pull/207/commits/06a33f311c627228536f5e8963aa56da173eeabc).